### DISCUSSION TITLE
excel file as input

### DISCUSSION THREAD

**Discussion by jeannepicard** [thelfer/tfel#560] [Q&A]
Hello, 
I have implemented a law in mfront tested with mtest and I am trying to have an excel file (or txt if needed) with different colomns as the inputs as the material property of my law. The first column of the excel file is the time and the others are the different parameters varying in function of time. 
I do not know what method to use to achieve this if someone could point me to the right direction I would really appreciate it. 
Sorry if this question as been answered before or is very obvious but I have not found the answer. 
Thank you very much for your help

**Comment by thelfer** [comment]
Hi @jeannepicard,

You can do this by adapting the following example:

```python
@Behaviour 'src\Release\Behaviour.dll' 'Elasticity';

@Evolution<data> 'YoungModulusEvolution' 'test.txt' using 1:2;
@MaterialProperty<function> 'YoungModulus' 'YoungModulusEvolution';
@MaterialProperty<constant> 'PoissonRatio' 0.3;
@ImposedStrain 'EXX' {0: 0, 1: 1e-2};
@ExternalStateVariable 'Temperature' 293.15;

@Times {0, 1};
```

where `test.txt` is a column file. In my test:

```
0 150e9
1 200e9
```

**Reply by thelfer** [reply]
Hi @jeannepicard, may I close this thread ?

**Comment by jeannepicard** [comment]
Hello Mr Helfer, yes thank you very much for your help, I am sorry I
thought I had sent a response.
Have a nice day

Le mer. 19 juin 2024 à 20:17, Thomas Helfer ***@***.***> a
écrit :
